core/src/main/c/share/ooo.cpp

```diff
@@ -422,6 +427,204 @@ typedef struct {
+            dest[0].i = src_pos | (1ull << 63);
+            dest++;
+            src_pos++;
+        } else if (src[src_pos] > index[index_pos].ts) {
+            dest[0] = index[index_pos];
+            dest++;
+            index_pos++;
+        } else {
+            // index_ts == src_ts
+            const uint64_t conflict_ts = src[src_pos];
+            while (index_pos <= index_hi_incl && index[index_pos].ts == conflict_ts) {
+                index_pos++;
+            }
+
+            // replace all records with same timestamp with last version from index
+            while (src_pos <= src_hi_incl && src[src_pos] == conflict_ts) {
+                dest[0] = index[index_pos - 1];
+                dest++;
+                src_pos++;
+            }
+        }
+    }
+
+    while (index_pos <= index_hi_incl) {
+        dest[0] = index[index_pos];
+        dest++;
+        index_pos++;
+    }
+
+    while (src_pos <= src_hi_incl) {
+        dest[0].ts = src[src_pos];
+        dest[0].i = src_pos | (1ull << 63);
+        dest++;
+        src_pos++;
+    }
+    return dest - dest_start;
+}
+
+int64_t merge_dedup_long_index_int_keys(
+        const uint64_t *src,
+        const int64_t src_lo,
+        const int64_t src_hi_incl,
+        const index_t *index,
+        const int64_t index_lo,
+        const int64_t index_hi_incl,
+        index_t *dest_start,
+        const int64_t dedupKeyCount,
+        const int32_t **dedupSrc,
+        const int64_t *dedupSrcTops,
+        const int32_t **dedupIndex
+) {
+    int64_t src_pos = src_lo;
+    int64_t index_pos = index_lo;
+    index_t *dest = dest_start;
+
+    uint64_t src_ts;
+    uint64_t index_ts;
+
+    while (src_pos <= src_hi_incl && index_pos <= index_hi_incl) {
+        src_ts = src[src_pos];
+        index_ts = (int64_t) index[index_pos].ts;
+
+        if (src_ts < index_ts) {
+            dest[0].ts = src_ts;
+            dest[0].i = src_pos | (1ull << 63);
+            dest++;
+            src_pos++;
+        } else if (src_ts > index_ts) {
+            dest[0].ts = index_ts;
+            dest[0].i = index_pos;
+            dest++;
+            index_pos++;
+        } else {
+            // index_ts == src_ts
+            const uint64_t conflict_ts = src_ts;
+            const int64_t conflict_src_pos = src_pos;
+            index_t *conflict_dest_start = dest;
+
+            // copy all src records with the same ts
+            while (src[src_pos] == conflict_ts) {
+                dest[0].ts = conflict_ts;
+                dest[0].i = src_pos | (1ull << 63);
+                dest++;
+                src_pos++;
+            }
+
+            // overwrite them with dest if key match
+            while (index[index_pos].ts == conflict_ts) {
+                bool matched = false;
+                for (int64_t i = conflict_src_pos; i < src_pos; i++) {
+                    if (key_equals(dedupKeyCount, i, dedupSrc, dedupSrcTops, index_pos, dedupIndex)) {
```

Comment:
nit: could you align parameter abd variable names?

Reply:
I removed this code, it's not used for timestamp dedup.